app/src/main/java/com/example/cocktails/CustomItem/UserItemLevel1.kt

```diff
@@ -363,9 +362,53 @@ class UserItemLevel1 : AppCompatActivity() {
         return super.dispatchTouchEvent(event)
     }
 
-}
+    private fun checkReadExternalStoragePermission() {
+        if (checkManifestReadExternalStoragePermission()) {
+            val intent = Intent()
+            intent.type = "image/*"
+            intent.action = Intent.ACTION_GET_CONTENT
+            startActivityForResult(intent, REQUEST_CODE_UPLOAD_IMG)
+        } else {
+            requestLocationPermissions()
+        }
+    }
+
+    private fun checkManifestReadExternalStoragePermission(): Boolean {
+        return ActivityCompat.checkSelfPermission(
+            this,
+            Manifest.permission.READ_EXTERNAL_STORAGE
+        ) == PackageManager.PERMISSION_GRANTED
+    }
+
+    private fun requestLocationPermissions() {
+        ActivityCompat.requestPermissions(
+            this,
+            arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE), PERMISSION_EXTERNAL_STORAGE_ID
+        )
+    }
 
-// Extension function to show toast message
-fun Context.toast(message: String) {
-    Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
+    @Override
+    override fun onRequestPermissionsResult(
+        requestCode: Int, permissions: Array<String?>,
+        grantResults: IntArray
+    ) {
+        when (requestCode) {
+            PERMISSION_EXTERNAL_STORAGE_ID -> {
+                if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    upload_img_Button.performClick()
+                } else {
+                    if (ActivityCompat.shouldShowRequestPermissionRationale(
+                            this, Manifest.permission.READ_EXTERNAL_STORAGE
+                        )
+                    ) {
+                        Toast.makeText(
+                            applicationContext,
+                            "you can't upload an image without location permission",
+                            Toast.LENGTH_LONG
```

Comment:
@enavhasson  "you can't upload an image without location permission" - why? or by mistake?

Reply:
By mistake 😅